docs/analyse.md

```diff
@@ -40,10 +40,307 @@ watch --interval 30 python -m neps.status ROOT_DIRECTORY
+            },
+        )
+        ################## TensorBoard Logging End ##################
+        scheduler.step()
+        
+        print(f"  Epoch {i + 1} / {max_epochs} Train Error: {avg_loss} ")
+
+    # Retrieve and return the information from the run pipeline.
+    return {
+        "loss": loss,
+        "info_dict": {
+            "cost": max_epochs,
+        },
+    }
+
+
+# Running the search with random search.
+if __name__ == "__main__":
+    set_seed(112)
+    logging.basicConfig(level=logging.INFO)
+
+    neps.run(
+        run_pipeline=run_pipeline,
+        pipeline_space=pipeline_space(),
+        root_directory="results/mnist_logging_example",
+        searcher="random_search",
+        max_evaluations_total=5,
+    )
+
+```
+
+#### Visualization Results
+
+The following command will open a local host for TensorBoard visualizations, allowing you to view them either in real-time or after the run is complete.
+
+```bash
+tensorboard --logdir path/to/root_directory
+```
+
+!!! Note
+    The annotations within the images provide insights into when and how each graph is triggered.
+
+This image shows visualizations related to scalar values logged during training. Scalars typically include metrics such as loss, incumbent trajectory, a summary of losses for all configurations, and any additional data provided via the `extra_data` argument in the `tblogger.log` function. 
+
+![scalar_loggings](doc_images/tensorboard/tblogger_scalar.jpg)
+
+This image represents visualizations related to logged images during training. It could include snapshots of input data, model predictions, or any other image-related information. In our case, we use images to depict instances of incorrect predictions made by the model.
+
+![image_loggings](doc_images/tensorboard/tblogger_image.jpg)
+
+The following images showcase visualizations related to hyperparameter logging in TensorBoard. These plots include three different views, providing insights into the relationship between different hyperparameters and their impact on the model.
+
+In the table view, you can explore hyperparameter configurations across five different trials. The table displays various hyperparameter values alongside corresponding evaluation metrics.
+
+![hparam_loggings1](doc_images/tensorboard/tblogger_hparam1.jpg)
+
+The parallel coordinate plot offers a holistic perspective on hyperparameter configurations. By presenting multiple hyperparameters simultaneously, this view allows you to observe the interactions between variables, providing insights into their combined influence on the model.
+
+![hparam_loggings2](doc_images/tensorboard/tblogger_hparam2.jpg)
+
+The scatter plot matrix view provides an in-depth analysis of pairwise relationships between different hyperparameters. By visualizing correlations and patterns, this view aids in identifying key interactions that may influence the model's performance.
+
+![hparam_loggings3](doc_images/tensorboard/tblogger_hparam3.jpg)
```

Comment:
These are very cool additions.

One minor nitpick is that it is not super clear what the text highlights mean on the images.

Secondly, if we are putting up images on the doc, can we do a proper run with more parameter ranges and budgets, and then put the images from that run? 
That should also give us a good scatter plot image, and a more wholesome set of learning curves.

Right now, it really looks like a poor man's HPO run and can be misinterpreted as not able to handle scale! 😅

Reply:
I removed the text highlights, it seemed all in all a cleaner look not include them. Linked the example and specified the modifications needed to be done to obtain the visualization results that are now available.

Used a more sophisticated HPO method + Images seem to be more meaningful.